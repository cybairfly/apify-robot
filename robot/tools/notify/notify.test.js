const { formatMessage } = require('./');
const { RobotError } = require('../../errors');


describe('formatMessage', () => {
  beforeEach(() => {
    process.env.APIFY_ACTOR_RUN_ID = 123;
  })
  test('messages match expected format', () => {
    const error = new RobotError('generic error message'); 
    const errorDetails = JSON.stringify(error, null, 4);
    const input = { target: 'hulu', debug: true, session: null, stealth: true }
    expect(formatMessage({input, error, errorLabel: error.type || error.name, errorDetails })).toMatchSnapshot();
  })
});
