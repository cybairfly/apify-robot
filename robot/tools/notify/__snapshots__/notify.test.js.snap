// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatMessage messages match expected format 1`] = `
"Error: hulu \`RobotError\`
https://my.apify.com/view/runs/123
\`\`\`{
    \\"0\\": \\"g\\",
    \\"1\\": \\"e\\",
    \\"2\\": \\"n\\",
    \\"3\\": \\"e\\",
    \\"4\\": \\"r\\",
    \\"5\\": \\"i\\",
    \\"6\\": \\"c\\",
    \\"7\\": \\" \\",
    \\"8\\": \\"e\\",
    \\"9\\": \\"r\\",
    \\"10\\": \\"r\\",
    \\"11\\": \\"o\\",
    \\"12\\": \\"r\\",
    \\"13\\": \\" \\",
    \\"14\\": \\"m\\",
    \\"15\\": \\"e\\",
    \\"16\\": \\"s\\",
    \\"17\\": \\"s\\",
    \\"18\\": \\"a\\",
    \\"19\\": \\"g\\",
    \\"20\\": \\"e\\",
    \\"name\\": \\"Robot.Error\\",
    \\"stealth\\": true,
    \\"debug\\": true,
    \\"session\\": null,
    \\"type\\": \\"RobotError\\",
    \\"retry\\": false
}\`\`\`"
`;
